client/gutenberg/editor/index.js

```diff
@@ -1,11 +1,28 @@
 import page from 'page';
-import { makeLayout, render as clientRender } from 'calypso/controller';
+import { makeLayout, redirectLoggedOut, render as clientRender } from 'calypso/controller';
+import { getSiteFragment } from 'calypso/lib/route';
 import { siteSelection, sites } from 'calypso/my-sites/controller';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getImmediateLoginLocale } from 'calypso/state/immediate-login/selectors';
 import { authenticate, post, redirect, siteEditor, exitPost } from './controller';
 
+function redirectToPermalinkIfLoggedOut( context, next ) {
+	if ( isUserLoggedIn( context.store.getState() ) ) {
+		return next();
+	}
+	const state = context.store.getState();
+	const siteFragment = context.params.site || getSiteFragment( context.path );
+	const login_locale = getImmediateLoginLocale( state );
```

Comment:
Seems like this will only work with immediate login. Is that intended? 

Also, do we need any additional checks - like whether it exists?

Reply:
I am not really sure if this is the right approach here. Ideally, we redirect the user to the correct locale. 
`getImmediateLoginLocale( state );` call is something that is done in the `isUserLoggedIn` function. So I kept it but I am not sure how to use it.

Reply:
I removed the call since I don't think its applicable.